From 9971eb532dac3efd71d9fe3e50e3d5fa20cee015 Mon Sep 17 00:00:00 2001
From: Dan Kalowsky <daniel.kalowsky@intel.com>
Date: Fri, 12 Sep 2014 12:58:47 -0700
Subject: [PATCH] Reduce D3 entry time

This patch reduced the D3 entry time on BYT-T-CR platforms from 10ms to
3ms, which can be handled by the system hardware.  This should provide
both a power and performance benefit through faster D3 entry and exit
times.

Change-Id: I08a5729dec28799910fc9d4a91ccfdbd955d34da
Signed-off-by: Dan Kalowsky <daniel.kalowsky@intel.com>
---
 drivers/pci/quirks.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 5cb726c..0ecf653 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -2972,6 +2972,21 @@ DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x8c26, quirk_remove_d3_delay);
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x8c4e, quirk_remove_d3_delay);
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x8c02, quirk_remove_d3_delay);
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x8c22, quirk_remove_d3_delay);
+/*
+ * PCI devices which are on Intel BYT-T systems can skip the 10ms delay before
+ * entering D3 mode, but do require a delay. In this case, 3ms is enough time
+ * properly move the devices around
+*/
+static void quirk_remove_byt_d3_delay(struct pci_dev *dev)
+{
+	dev->d3_delay = 3;
+	dev->d3cold_delay = 3;
+}
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x0f00, quirk_remove_byt_d3_delay);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x0f35, quirk_remove_byt_d3_delay);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x0f37, quirk_remove_byt_d3_delay);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x0f18, quirk_remove_byt_d3_delay);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x0f1c, quirk_remove_byt_d3_delay);
 
 /*
  * Some devices may pass our check in pci_intx_mask_supported if
-- 
1.9.3

